android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java

```diff
@@ -167,15 +167,20 @@ private void nextOnboardingStep() {
             int margin = mIsTablet ? 100 : 0;
             setLeafAnimation(mVLeafAlignTop, mIvLeafTop, 1f, margin, true);
             setLeafAnimation(mVLeafAlignBottom, mIvLeafBottom, 1f, margin, false);
-            setFadeInAnimation(mTvWelcome, 200);
+            if (mTvWelcome != null) {
```

Comment:
I don't think that fully fixes of what we have here. `mTvWelcome` as well as `mVLeafAlignTop`, `mVLeafAlignBottom`, `mIvLeafTop` and `mIvLeafBottom` are been initialized inside `initViews`, but we had a crash on accessing `mTvWelcome` only. That tells us that `initViews` and `nextOnboardingStep ` are not synchronized and we can have crashes on
`setLeafAnimation(mVLeafAlignTop, mIvLeafTop, 1f, margin, true);` and on
`setLeafAnimation(mVLeafAlignBottom, mIvLeafBottom, 1f, margin, false);`
I think we need to do checks for `mVLeafAlignTop`, `mVLeafAlignBottom`, `mIvLeafTop` and `mIvLeafBottom` as well or even better synchronize `initViews` and `nextOnboardingStep`. As it looks like next step has been called while init isn't fully done yet. We should prevent doing that. I think null checks for now will solve it and we will see how it performs.

Reply:
Yes, looks like synchronize issue between `initializeViews` and `finishNativeInitialization`. As per code it should work properly. Let me try to understand and check chromium code